src/riot-rs-threads/src/arch/cortex_m.rs

```diff
@@ -62,65 +62,11 @@ impl Arch for Cpu {
     }
 
     #[inline(always)]
-    fn start_threading(next_sp: usize) {
-        cortex_m::interrupt::disable();
+    fn start_threading() {
         Self::schedule();
-        unsafe {
-            asm!(
-                "
-                msr psp, r1 // set new thread's SP to PSP
-                cpsie i     // enable interrupts, otherwise svc hard faults
-                svc 0       // SVC 0 handles switching
```

Comment:
Could you briefly explain why SVCall was previously needed, but not anymore?

Reply:
The first iterations of PendSV didn't handle the "initial" case, coming from no thread. At least I thought so, and in RIOT-c that is still the case. I tried hard not to special-case this, as the scheduler gets called potentially billions of times in the regular case and only once in the "initial" case. So even a single extra "if" or cmp is a waste.
Using a minimal SVC that handles switching to the first thread seemed like an easy solution.

When doing this PR, I realized that actually PendSV *does* handle the no-thread case properly now.